contracts/staking/StakingPool.sol

```diff
@@ -0,0 +1,453 @@
+            "StakingPool: pool full"
+        );
+        _;
+    }
+
+    function deposit(uint256 amount)
+        external
+        stakingPeriodNotStarted
+        nonReentrant
+        stakingPoolNotFull(amount)
+    {
+        console.log("depositing..!");
+        require(
+            amount >= _stakingPoolInfo.minimumContribution,
+            "StakingPool: min contribution"
+        );
+
+        User storage user = _users[_msgSender()];
+
+        user.depositAmount += uint128(amount);
+        _stakingPoolInfo.totalStakedAmount += uint128(amount);
+
+        StakingPoolLib.StakingPoolType poolType = _stakingPoolInfo.poolType;
+
+        for (uint256 i = 0; i < _stakingPoolInfo.rewardTokens.length; i++) {
+            uint256 rewardsPerShare = 0;
+            if (poolType == StakingPoolLib.StakingPoolType.FLOATING) {
+                _computeRewardsPerShare(i);
+                _stakingPoolInfo.rewardTokens[i].rewardAmountRatio = uint32(
+                    rewardsPerShare
+                );
+            } else {
+                // fixed
+                user.rewardAmounts[i] += _computeRewards(uint128(amount), i);
+            }
+
+            // assign storage for rewards amount
+            user.rewardAmounts.push(0);
+        }
+
+        emit Deposit(_msgSender(), amount);
+
+        require(
+            _stakingPoolInfo.stakeToken.transferFrom(
+                _msgSender(),
+                address(this),
+                amount
+            ),
+            "StakingPool: failed to transfer"
+        );
+    }
+
+    function withdraw()
+        external
+        stakingPeriodComplete
+        rewardsAvailable
+        nonReentrant
+    {
+        User memory user = _users[_msgSender()];
+
+        console.log("user.depositAmount ", user.depositAmount);
+
+        require(user.depositAmount > 0, "StakingPool: not eligible");
+
+        delete _users[_msgSender()];
+        emit Withdraw(_msgSender(), user.depositAmount);
+        require(
+            _stakingPoolInfo.stakeToken.transfer(
+                _msgSender(),
+                uint256(user.depositAmount)
+            ),
+            "StakingPool: stake tx fail"
+        );
+
+        StakingPoolLib.StakingPoolType poolType = _stakingPoolInfo.poolType;
+
+        for (uint256 i = 0; i < _stakingPoolInfo.rewardTokens.length; i++) {
+            uint256 amount = 0;
+
+            // floating
+            if (poolType == StakingPoolLib.StakingPoolType.FLOATING) {
+                amount = uint256(
+                    _stakingPoolInfo.rewardTokens[i].rewardAmountRatio *
+                        user.depositAmount
+                );
+            } else {
+                // fixed
+                amount = uint256(user.rewardAmounts[i]);
+            }
+
+            IERC20 token = IERC20(_stakingPoolInfo.rewardTokens[i].token);
+
+            emit WithdrawRewards(_msgSender(), address(token), amount);
+
+            require(
```

Comment:
@CjHare This is an example of Slither not liking external calls in a loop.

Reply:
At least it's being consistent about not liking calls in loops 😁